ovs-p4rt/ovs_p4rt.cc

```diff
@@ -619,10 +744,91 @@ void PrepareEncapTableEntry(p4::v1::TableEntry* table_entry,
+                                  bool insert_entry) {
+  table_entry->set_table_id(GetTableId(p4info, GENEVE_ENCAP_MOD_TABLE));
+  auto match = table_entry->add_match();
+  match->set_field_id(
+      GetMatchFieldId(p4info, GENEVE_ENCAP_MOD_TABLE,
+                      GENEVE_ENCAP_MOD_TABLE_KEY_VENDORMETA_MOD_DATA_PTR));
+  match->mutable_exact()->set_value(EncodeByteValue(1, tunnel_info.vni));
+
+  if (insert_entry) {
+    auto table_action = table_entry->mutable_action();
+    auto action = table_action->mutable_action();
+    action->set_action_id(GetActionId(p4info, ACTION_GENEVE_ENCAP));
+    {
+      auto param = action->add_params();
+      param->set_param_id(GetParamId(p4info, ACTION_GENEVE_ENCAP,
+                                     ACTION_GENEVE_ENCAP_PARAM_SRC_ADDR));
+      param->set_value(CanonicalizeIp(tunnel_info.local_ip.ip.v4addr.s_addr));
+    }
+    {
+      auto param = action->add_params();
+      param->set_param_id(GetParamId(p4info, ACTION_GENEVE_ENCAP,
+                                     ACTION_GENEVE_ENCAP_PARAM_DST_ADDR));
+      param->set_value(CanonicalizeIp(tunnel_info.remote_ip.ip.v4addr.s_addr));
+    }
+    {
+      auto param = action->add_params();
+      param->set_param_id(GetParamId(p4info, ACTION_GENEVE_ENCAP,
+                                     ACTION_GENEVE_ENCAP_PARAM_SRC_PORT));
+      uint16_t dst_port = htons(tunnel_info.dst_port);
+
+      param->set_value(EncodeByteValue(2, (((dst_port * 2) >> 8) & 0xff),
+                                       ((dst_port * 2) & 0xff)));
+    }
+    {
+      auto param = action->add_params();
+      param->set_param_id(GetParamId(p4info, ACTION_GENEVE_ENCAP,
+                                     ACTION_GENEVE_ENCAP_PARAM_DST_PORT));
+      uint16_t dst_port = htons(tunnel_info.dst_port);
+
+      param->set_value(
+          EncodeByteValue(2, ((dst_port >> 8) & 0xff), (dst_port & 0xff)));
+    }
+    {
+      auto param = action->add_params();
+      param->set_param_id(GetParamId(p4info, ACTION_GENEVE_ENCAP,
+                                     ACTION_GENEVE_ENCAP_PARAM_VNI));
+      param->set_value(EncodeByteValue(1, tunnel_info.vni));
+    }
+  }
+
+  return;
+}
+#endif
+
+void PrepareEncapTableEntry(p4::v1::TableEntry* table_entry,
+                            const struct tunnel_info& tunnel_info,
+                            const ::p4::config::v1::P4Info& p4info,
+                            bool insert_entry) {
+#if defined(DPDK_TARGET)
+  PrepareVxlanEncapTableEntry(table_entry, tunnel_info, p4info, insert_entry);
+#elif defined(ES2K_TARGET)
+  if (tunnel_info.tunnel_type == OVS_TUNNEL_VXLAN) {
+    PrepareVxlanEncapTableEntry(table_entry, tunnel_info, p4info, insert_entry);
+  } else if (tunnel_info.tunnel_type == OVS_TUNNEL_GENEVE) {
+    PrepareGeneveEncapTableEntry(table_entry, tunnel_info, p4info,
+                                 insert_entry);
+  } else {
+    std::cout << "ERROR: Unsupported tunnel type" << std::endl;
```

Comment:
Is there a logger we can send this to?

there is no console to look at for debugging in case of ovs-sidecar and dont think this serves any purpose